File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/MallpayOrder.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects;

public class MallpayOrder : ApiBase
{
    [JsonProperty("totalPrice")] public MallpayPrice? TotalPrice { get; set; }

    [JsonProperty("totalVat")] public List<MallpayVat>? TotalVat { get; set; } = new();

    [JsonProperty("addresses")] public List<MallpayAddress>? Addresses { get; set; } = new();


    [JsonProperty("deliveryType")] public string? DeliveryType { get; set; }

    [JsonProperty("carrierId")] public string? CarrierId { get; set; }

    [JsonProperty("carrierCustom")] public string? CarrierCustom { get; set; }

    [JsonProperty("items")] public List<MallpayOrderItem>? Items { get; set; }


    public string ToSign()
    {
        var sb = new StringBuilder();
        if (TotalPrice != null) Add(sb, TotalPrice.ToSign());
        if (TotalVat != null)
            foreach (var vat in TotalVat)
                Add(sb, vat.ToSign());
        if (Addresses != null)
            foreach (var address in Addresses)
                Add(sb, address.ToSign());
        Add(sb, DeliveryType);
        Add(sb, CarrierId);
        Add(sb, CarrierCustom);
        if (Items != null)
            foreach (var item in Items)
                Add(sb, item.ToSign());
        DeleteLast(sb);
        return sb.ToString();
    }
}

File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/MallpayOrder.java
package cz.monetplus.mips.eapi.v19.connector.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@EqualsAndHashCode(callSuper = true)
@JsonInclude(value = JsonInclude.Include.NON_NULL)
@AllArgsConstructor @NoArgsConstructor
public class MallpayOrder extends ApiBase implements Signable {
    private MallpayPrice totalPrice;
    private List<MallpayVat> totalVat = new ArrayList<>();
    private List<MallpayAddress> addresses = new ArrayList<>();
    private String deliveryType;
    private String carrierId;
    private String carrierCustom;
    private List<MallpayOrderItem> items;

    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        if(totalPrice != null) add(sb, totalPrice.toSign());
        if(totalVat != null) for (MallpayVat vat : totalVat) add(sb, vat.toSign());
        if(addresses != null) for (MallpayAddress address : addresses) add(sb, address.toSign());
        add(sb, deliveryType);
        add(sb, carrierId);
        add(sb, carrierCustom);
        if(items != null) for (MallpayOrderItem item : items) add(sb, item.toSign());
        deleteLast(sb);
        return sb.toString();
    }
}
